wordpress-mobile/WordPress-FluxC-Android
# Fix auth-options test failing on the release stack

First, some context: the `testFetchAuthOptionsForUserWithUnverifiedEmail()` test was failing on the release stack with an `EMAIL_LOGIN_NOT_ALLOWED` error, which asks for an username to be provided instead of an email. This can happen under certain circumstances involving an unverified email as a security measure by the `users/%s/auth-options` endpoint, which will classify that email as "suspicious".

This PR does two things to address that:
1. It moves the `testFetchAuthOptionsForUserWithUnverifiedEmail()` test to the mocked stack.
1. It adds a new `testFetchAuthOptionsForUserWithSuspiciousEmail()` test to the mocked stack, in order to properly handle the `EMAIL_LOGIN_NOT_ALLOWED` error.

## To test

You can either run `MockedStack_AccountTest` and verify all tests pass or use the following steps on the example app:
1. On the initial screen, tap **Signed Out Actions**.
1. On the next screen, tap **Fetch Auth Options**.
1. On the dialog, enter an unverified email address and tap **Ok**.
1. Verify the app logs a successful response in the console and that the field `Is email verified` is `false`.
1. Tap **Fetch Auth Options** again.
1. On the dialog, enter an email address that is classified as "suspicious" and tap **Ok**.
1. Verify the app logs a failed response in the console and that the error is `EMAIL_LOGIN_NOT_ALLOWED`.
